src/Analyzer.Cli.FunctionalTests/ValidTemplateConstants.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Analyzer.Cli.FunctionalTests
+{
+    public class ValidTemplateConstants
+    {
+        public const string MissingStartObject = @" ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#""}";
+
+        public const string NoValidTopLevelProperties =@"
+            { 
+              ""parameters"": {
+              ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#""
+            }    
+        }";
+
+        public const string CaseSensitivity = @"{""$schEmA"": ""https://ScHeMa.mAnAgEmEnt.AzUrE.cOm/schEmAs/2019-04-01/deploymentTemplate.json#""}";
+
+        public const string MissingSchema = @"{""contentVersion"": ""1.0.0.0"", ""Parameters"": {}}";
+
+        public const string SchemaValueNotString = @"{ ""$schema"": 5 }";
+
+        public const string NoSchemaInvalidProperties = @" {""properties"": }";
+
+        public const string DifferentSchemaDepths = @"
+        {
+            ""parameters"": ""{
+              ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#""
```

Comment:
I believe the idea of this test is to demonstrate that the inner schema is ignored, and the outer schema is what is actually honored.  However, because they are the same thing, there is no way to verify in the test which value was used.  To test this properly, the inner schema should be set to an invalid value so that if the wrong one is taken, the test fails.

Additionally, continuing from my other comment, this should be made into a valid template so it's clear that `IsValidTemplate` returned correctly.